common/functions/src/scalars/strings/char_.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct CharFunction {
+    _display_name: String,
+}
+
+impl CharFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(CharFunction {
+            _display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for CharFunction {
+    fn name(&self) -> &str {
+        "char"
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        Some((1, 1024))
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::String)
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, _input_rows: usize) -> Result<DataColumn> {
+        let len = columns[0].column().len();
+        let mut l_array = DFStringArray::full(&[], len);
+
+        for column in columns.iter() {
+            let column = column.column();
+            if column.data_type().is_null() {
+                continue;
+            }
+            let column = column.cast_with_type(&DataType::UInt32)?;
+            match column {
+                DataColumn::Array(r_array) => {
+                    let r_array = r_array.u32()?;
+                    l_array = DFStringArray::new_from_iter_validity(
+                        l_array
+                            .into_no_null_iter()
+                            .zip(r_array.into_no_null_iter())
```

Comment:
Too much mem `allocation` and `memcpy` here.

Reply:
We can use `DFStringArray::from_data_unchecked` to build result columns.

For values buffer, we can reserve the size to be `rows * args_size`.

I contributed the `char` function of ClickHouse before, hope this can help you.

https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/FunctionChar.cpp#L55-L91

Reply:
Thanks for you review.
I have read your implementation of `char()` function. It seems there are some definition differences between your char() function and mysql8's. In mysql8's definition, one integer can be converted into 4 chars at most. So which definition is the one to implement? 
If mysql8's definition is the one to implement, my solution is to reserve a values buffer, the size of which is `4 * rows * args_size`.

Reply:
>  one integer can be converted into 4 chars at most

I did not get it. Integer will be cast into u8, so it just took one char.

```
MySQL> select char(64);
+----------+
| char(64) | 
+----------+
| @        | 
+----------+
1 row in set (0.00 sec)
MySQL> select char(64 + 256) ;
+-----------------+
| char(64 + 256)  | 
+-----------------+
| @              | 
+-----------------+
1 row in set (0.00 sec)
```

Reply:
How about the parameter type? Do we only support numeric type?

Reply:
Yes, we can use `cast_with_type(&DataType::UInt8)` to cast all parameter columns.

Reply:
Is there any method to generate a bitmap full of 1? One parameter's type of function `DFStringArray::from_data_unchecked` is `Option<Bitmap>`

Reply:
Just use None!